docs/apis/plugintypes/repository/index.mdx

```diff
@@ -0,0 +1,975 @@
+
+In order to access to a flickr public account, the plugin needs to know the email address of the Flickr public account owner. So the administrator will need to set an email address for every repository. Let's add an "email address" setting to every repository. To do so you need to override:
+
+- **get_instance_option_names** returing ['email_address'].
+- **instance_config_form** adding a text box called 'email_address' into the form.
+
+So at this moment all our Flickr Public Repositories will have a specific email address. However this is not enough. In order to communicate with Flickr, Moodle needs to know a Flickr API key (<http://www.flickr.com/services/api/>). This API key is the same for any repository. We could add it with the email address setting but the administrator would have to enter the same API key for every repository. Hopefully the administrator can add settings to the plugin level, impacting all repositories. To do so you need to override:
+
+- **get_type_option_names** returing ['api_key'].
+- **type_config_form** adding the api_key text input element into the form.
+
+At this point we have created everything necessary for the administration pages. But let's go further. It would be good if the user can enter any "Flickr public account email address" in the file picker. In fact we want to display in the file picker a Flickr Public repository that the Moodle administrator can never delete. Let's add:
+
+- **plugin_init** using __repository::static_function__ to create a default repository instance.
+
+That's all - the administration part of our Flickr Public plugin is done. For your information, Box.net, Flickr, and Flickr Public all have similar administration APIs.
+
+<details>
+  <summary>View the full example code</summary>
+  <div>
+
+import FlickPublicLib from '!!raw-loader!./_examples/flickrpublic_lib.php';
+
+<CodeBlock language="js" title="repository/flickrpublic/lib.php">{FlickPublicLib}</CodeBlock>
+
+  </div>
+</details>
+
+## Repository APIs
+
+### Quick Start
+
+First of all, the File Picker using intensively Ajax you will need a easy way to debug.
+
+- Our first question when you write your plugin specification is 'Does the user need to log-in'? If they do, in your plugin you have to detect user session in constructor() function, and use print_login() if required, see more details below.
+- For most of plugins, you need to establish a connection with the remote repository. This connection can be done into the get_listing(), constructor() function, see more details below.
+- You want to retrieve the file that the user selected, rewrite get_file() if required, see more details below.
+- Optional question that you should ask yourself is 'Does the user can execute a search', if they do, you will have to rewrite search() method, see more details below.
```

Comment:
This section is a very different writing style to the previous. Much more colloqial

Reply:
Thanks. Fixed.